src/core/utils/helpers/general.ts

```diff
@@ -217,6 +217,42 @@ export const sortByReservationDate = (list: ReservationType[]) => {
   );
 };
 
+export const sortByDueDate = (list: LoanType[]) => {
+  //  We use orderBy from lodash to avoid mutating the original list
+  return orderBy(
+    list,
+    (item) =>
+      item.dueDate ? dayjs(item.dueDate).startOf("day").valueOf() : Infinity,
```

Comment:
I’ve removed startOf("day") as suggested. Initially, I thought it would be a good idea to normalize the dates by stripping out the time component to ensure consistent sorting, especially when the underlying data includes both dates and times.